src/PeriodEnd/Refunds/SFA.DAS.ProviderPayments.Calc.Refunds.UnitTests/ServiceTests/GivenALearnerRefundProcessor/AndThereArePreviousPaymentsForTheMonth.cs

```diff
@@ -0,0 +1,373 @@
+
+                actual.Where(x => x.TransactionType == TransactionType.Balancing).Sum(x => x.Amount).Should().BeApproximately(-900, 0.00005m);
+                actual.Where(x => x.TransactionType == TransactionType.Balancing16To18FrameworkUplift).Sum(x => x.Amount).Should().BeApproximately(-1200, 0.00005m);
+            }
+
+            [Test, RefundsAutoData]
+            public void AndPaymentsNotSufficient_ThenTheRefundPaymentsAreTheSameAsTheMatchingPreviousPayments(
+                LearnerRefundProcessor sut)
+            {
+                var data = RefundGenerator.Generate(numberOfRefunds: 2, refundAmount: -900, paymentAmount: 200);
+
+                var refundOne = data.Refunds[0];
+                refundOne.TransactionType = TransactionType.Balancing;
+                refundOne.AmountDue = -900;
+                var refundTwo = data.Refunds[1];
+                refundTwo.TransactionType = TransactionType.Balancing16To18FrameworkUplift;
+                refundTwo.AmountDue = -1200;
+
+                data.AssociatedPayments[0].TransactionType = TransactionType.Balancing;
+                data.AssociatedPayments[1].TransactionType = TransactionType.Balancing;
+                data.AssociatedPayments[2].TransactionType = TransactionType.Balancing;
+
+                data.AssociatedPayments[3].TransactionType = TransactionType.Balancing16To18FrameworkUplift;
+                data.AssociatedPayments[4].TransactionType = TransactionType.Balancing16To18FrameworkUplift;
+                data.AssociatedPayments[5].TransactionType = TransactionType.Balancing16To18FrameworkUplift;
+                data.AssociatedPayments[3].Amount = 100;
+                data.AssociatedPayments[4].Amount = 100;
+                data.AssociatedPayments[5].Amount = 100;
+
+                var actual = sut.ProcessRefundsForLearner(data.Refunds, data.AssociatedPayments);
+
+                actual.Where(x => x.TransactionType == TransactionType.Balancing).Sum(x => x.Amount).Should().BeApproximately(-600, 0.00005m);
+                actual.Where(x => x.TransactionType == TransactionType.Balancing16To18FrameworkUplift).Sum(x => x.Amount).Should().BeApproximately(-300, 0.00005m);
+            }
+        }
+
+        [TestFixture]
+        public class AndThereAreMultipleRefundsWithDifferentContractTypes
+        {
+            [Test, RefundsAutoData]
+            public void AndPaymentsSufficient_ThenTheRefundPaymentsAreCorrect(
+                LearnerRefundProcessor sut)
+            {
+                var data = RefundGenerator.Generate(numberOfRefunds: 2, refundAmount: -900, paymentAmount: 500);
+
+                var refundOne = data.Refunds[0];
+                refundOne.ApprenticeshipContractType = 1;
+                var refundTwo = data.Refunds[1];
+                refundTwo.ApprenticeshipContractType = 2;
+                refundTwo.AmountDue = -1200;
+
+                data.AssociatedPayments[0].ApprenticeshipContractType = 1;
```

Comment:
This is a real side issue (but we have enums for the FundingSource and TransactionType), why not for teh apprenticeshipContractType (we don't need to resolve this now)